src/EnergyPlus/DataSurfaces.cc

```diff
@@ -180,7 +170,7 @@ Surface2D::Surface2D(ShapeCat const shapeCat, int const axis, Vertices const &v,
                 xt = xte;
             }
 #endif
-            assert((shapeCat == ShapeCat::Nonconvex) || (crossEdges.size() == 2));
+            assert((shapeCat == ShapeCat::Nonconvex) || (crossEdges.size() == 2u));
```

Comment:
This is the assert that was failing. Initially, I thought it was incorrect and should be `>=2`, but in the end the failing surface was `ShapeCat:Convex` here when it should have been `Nonconvex`.